crates/amaru-consensus/src/consensus/chain_selection.rs

```diff
@@ -275,23 +275,79 @@ where
 }
 
 #[cfg(test)]
-mod tests {
-
+pub(crate) mod tests {
     use super::*;
-    use crate::consensus::header::test::{generate_headers_anchored_at, random_bytes, TestHeader};
+    use amaru_kernel::{from_cbor, to_cbor};
+    use amaru_ouroboros_traits::is_header::fake::FakeHeader;
+    use proptest::prelude::*;
+    use rand::{rngs::StdRng, RngCore, SeedableRng};
+
+    /// Very simple function to generate random sequence of bytes of given length.
+    pub fn random_bytes(arg: u32) -> Vec<u8> {
+        let mut rng = StdRng::from_entropy();
+        let mut buffer = vec![0; arg as usize];
+        rng.fill_bytes(&mut buffer);
+        buffer
+    }
+
+    /// Generate a chain of headers anchored at a given header.
+    ///
+    /// The chain is generated by creating headers with random body hash, and linking
+    /// them to the previous header in the chain until the desired length is reached.
+    pub fn generate_headers_anchored_at(anchor: FakeHeader, length: u32) -> Vec<FakeHeader> {
+        let mut headers: Vec<FakeHeader> = Vec::new();
+        let mut parent = anchor;
+        for i in 0..u64::from(length) {
+            let header = FakeHeader::FakeHeader {
+                block_number: i + anchor.block_height() + 1,
+                slot: i + anchor.slot() + 1,
+                parent: parent.hash(),
+                body_hash: random_bytes(32).as_slice().into(),
+            };
+            headers.push(header);
+            parent = header;
+        }
+        headers
+    }
+
+    prop_compose! {
+        fn any_test_header()(
+            slot in 0..1000000u64,
+            block_number in 0..100000u64,
+            parent in any::<[u8; 32]>(),
+            body in any::<[u8; 32]>(),
+        )
+            -> FakeHeader {
+            FakeHeader::FakeHeader {
+                block_number,
+                slot,
+                parent: parent.into(),
+                body_hash: body.into(),
+            }
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn prop_roundtrip_cbor(hdr in any_test_header()) {
+            let bytes = to_cbor(&hdr);
+            let hdr2 = from_cbor::<FakeHeader>(&bytes).unwrap();
+            assert_eq!(hdr, hdr2);
+        }
+    }
```

Comment:
Note: this was moved from the `consensus::header::test` module here; while the definition of `TestHeader` has been moved to `ouroboros_trait` and renamed `FakeHeader`.